Marlin/src/inc/Conditionals_post.h

```diff
@@ -1936,206 +1928,281 @@
   #endif
 #endif
 
-//
-// Endstops and bed probe
-//
-
-#define _HAS_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M))
+/**
+ * Endstop and probe flags
+ * - Set USE_(AXIS)_(MIN|MAX) flags for each used endstop that has a pin, including those for DIAG0 state.
+ *   - Note: Dual X Carriage uses "X" and "X2" steppers, but X_MIN and X_MAX endstop states (i.e., not X2_MAX).
+ * - Set a HAS_(AXIS)_(MIN|MAX)_STATE flag for each endstop that has a state, including SPI Sensorless which don't use a pin.
+ * - Set a HAS_(AXIS)_STATE flag for each axis that has at least one state.
+ */
+#define _USE_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M) && !A##_SPI_SENSORLESS)
 
-#if _HAS_STOP(X,,MIN,)
-  #define HAS_X_MIN 1
+#if _USE_STOP(X,,MIN,)
```

Comment:
SENSORLESS_PROBING in a delta will also need USE_X_MIN and USE_Y_MIN. Any of the AXIS triggers the limit switch during polling.
I think it should be written as follows;
#if _USE_STOP(X,,MIN,ENABLED(HAS_DELTA_SENSORLESS_PROBING)))